### PR TITLE
Only update the envfile in __update_envfile

### PR DESCRIPTION
**Pull request by ptnapoleon**
@mambocab to review, while I run on CI. This fixes the relevant windows test, which is promising.

### CODE REVIEW

**Comment by ptnapoleon** [comment]
This breaks every single dtest, hold off while I rewrite the patch

**Comment by ptnapoleon** [comment]
Okay, review again, please. Running on CI but it appears to work now.

**Comment by ptnapoleon** [comment]
CI says this is worth reviewing. Thoughts @mambocab?

File: ccmlib/common.py [hunk 1]
```diff
@@ -185,19 +185,11 @@ def make_cassandra_env(install_dir, node_path, update_conf=True):
...
-                ('env:CASSANDRA_HOME =', '        $env:CASSANDRA_HOME="%s"' % install_dir),
-                ('env:CASSANDRA_CONF =', '    $env:CCM_DIR="' + node_path + '\\conf"\n    $env:CASSANDRA_CONF="$env:CCM_DIR"'),
-                ('cp = ".*?env:CASSANDRA_HOME.conf', '    $cp = """$env:CASSANDRA_CONF"""')
-            ]
-        else:
-            replacements = [
-                ('CASSANDRA_HOME=', '\tCASSANDRA_HOME=%s' % install_dir),
-                ('CASSANDRA_CONF=', '\tCASSANDRA_CONF=%s' % os.path.join(node_path, 'conf'))
-            ]
+    if update_conf and not (is_win() and get_version_from_build(node_path=node_path) >= '2.1'):
```
Comment (mambocab) [inline ccmlib/common.py]: This could do with some documenting comments and/or pulling out the guard logic into a variable.

File: ccmlib/node.py [hunk 2]
```diff
@@ -1417,6 +1417,16 @@ def __update_envfile(self):
 
         common.replace_in_file(conf_file, jmx_port_pattern, jmx_port_setting)
 
+
+        if common.is_win() and common.get_version_from_build(node_path=self.get_path()) >= '2.1':
```
Comment (mambocab) [inline ccmlib/node.py]: Same here -- there should be some documentation about the changes in C* that make this necessary.

**Comment by mambocab** [comment]
Re: my last 2 comments: since those use the same guard and has a very specific meaning, maybe make it a method on `Node`, like `node.cassandra_env_in_bin_dir()`. There are conditions where env vars are set somewhere special, and I think both of these bits of code would be clearer if you could explicitly ask the node that.

**Comment by mambocab** [comment]
If this patch makes stuff work again, merge away. I don't understand the need for these methods well enough to give a really confident +1, but that's expected.

**Comment by ptnapoleon** [comment]
Filed a cstar ticket to clean up the windows gating logic everywhere. Merging this since it's a horribly critical bug.

**Pull request merged**